src/plugins/sails.js

```diff
@@ -54,12 +54,21 @@ io.socket.on('game', function (evData) {
           store.commit('updateReady', evData.data.pNum);
           break;
         case 'Initialize': {
+          const currentRoute = router.currentRoute.value;
+          const isSpectating = currentRoute.name === ROUTE_NAME_SPECTATE;
+
+          // Update state
           store.commit('resetState');
           store.dispatch('updateGameThenResetPNumIfNull', evData.data.game);
-          const gameRoute = `/game/${store.state.game.id}`;
-          const currentRoute = router.currentRoute.fullPath;
-          if (gameRoute !== currentRoute) {
-            router.push(gameRoute);
+          if (isSpectating) {
+            store.commit('setMyPNum', 0); // always spectate as p0
+          }
+
+          // Validate current route & navigate if incorrect
+          const expectedRoutePrefix = isSpectating ? 'spectate' : 'game';
```

Comment:
These should use route constants if we have them (which I think we do). Might need to move them to a more local position so it can be shared though. If it's too much work or annoying you can make a ticket for it.

Reply:
Yeah it could also be done via route names instead of paths, which might be less brittle. I think we can do that in the same PR; shouldn't be too big a deal

Reply:
Agreed.